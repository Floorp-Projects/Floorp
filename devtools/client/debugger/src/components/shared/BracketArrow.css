/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

.bracket-arrow {
  position: absolute;
  pointer-events: none;
}

.bracket-arrow::before,
.bracket-arrow::after {
  content: "";
  height: 0;
  width: 0;
  position: absolute;
  border: 7px solid transparent;
}

.bracket-arrow.up::before {
  border-bottom-color: var(--theme-splitter-color);
  top: -1px;
}

.theme-dark .bracket-arrow.up::before {
  border-bottom-color: var(--theme-body-color);
}

.bracket-arrow.up::after {
  border-bottom-color: var(--theme-body-background);
  top: 0px;
}

.bracket-arrow.down::before {
  border-bottom-color: transparent;
  border-top-color: var(--theme-splitter-color);
  top: 0px;
}

.theme-dark .bracket-arrow.down::before {
  border-top-color: var(--theme-body-color);
}

.bracket-arrow.down::after {
  border-bottom-color: transparent;
  border-top-color: var(--theme-body-background);
  top: -1px;
}

.bracket-arrow.left::before {
  border-left-color: transparent;
  border-right-color: var(--theme-splitter-color);
  top: 0px;
}

.theme-dark .bracket-arrow.left::before {
  border-right-color: var(--theme-body-color);
}

.bracket-arrow.left::after {
  border-left-color: transparent;
  border-right-color: var(--theme-body-background);
  top: 0px;
  left: 1px;
}
