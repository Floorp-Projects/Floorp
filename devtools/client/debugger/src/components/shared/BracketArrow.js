/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import React from "react";
import classNames from "classnames";

import "./BracketArrow.css";

const BracketArrow = ({ orientation, left, top, bottom }) => {
  return (
    <div
      className={classNames("bracket-arrow", orientation || "up")}
      style={{ left, top, bottom }}
    />
  );
};

export default BracketArrow;
