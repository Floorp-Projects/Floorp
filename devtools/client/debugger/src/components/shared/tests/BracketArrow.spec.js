/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import React from "react";
import { shallow } from "enzyme";

import BracketArrow from "../BracketArrow";

describe("BracketArrow", () => {
  const wrapper = shallow(
    <BracketArrow orientation="down" left={10} top={20} bottom={50} />
  );
  it("render", () => expect(wrapper).toMatchSnapshot());
  it("render up", () => {
    wrapper.setProps({ orientation: null });
    expect(wrapper).toMatchSnapshot();
  });
});
