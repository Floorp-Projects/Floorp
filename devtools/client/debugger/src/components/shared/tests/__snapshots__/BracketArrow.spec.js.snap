// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BracketArrow render 1`] = `
<div
  className="bracket-arrow down"
  style={
    Object {
      "bottom": 50,
      "left": 10,
      "top": 20,
    }
  }
/>
`;

exports[`BracketArrow render up 1`] = `
<div
  className="bracket-arrow up"
  style={
    Object {
      "bottom": 50,
      "left": 10,
      "top": 20,
    }
  }
/>
`;
